crates/oxc_linter/src/rules/react_hooks/exhaustive_deps.rs

```diff
@@ -0,0 +1,3454 @@
+use oxc_span::{Atom, Span};
+use phf::phf_set;
+
+use crate::{
+    ast_util::{get_declaration_of_variable, get_enclosing_function},
+    context::LintContext,
+    rule::Rule,
+    AstNode,
+};
+
+fn missing_callback_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("React hook {hook_name} requires an effect callback."))
+        .with_label(span)
+        .with_help("Did you forget to pass a callback to the hook?")
+}
+
+fn dependency_array_required_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "React Hook {hook_name} does nothing when called with only one argument."
+    ))
+    .with_label(span)
+    .with_help("Did you forget to pass an array of dependencies?")
+}
+
+fn unknown_dependencies_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+      "React Hook {hook_name} received a function whose dependencies are unknown. Pass an inline function instead."
+    ))
+    .with_label(span)
+}
+
+fn async_effect_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Effect callbacks are synchronous to prevent race conditions.")
+        .with_label(span)
+}
+
+fn missing_dependency_diagnostic(hook_name: &str, deps: &[String], span: Span) -> OxcDiagnostic {
+    let deps_pretty = if deps.len() == 1 {
+        format!("'{}'", deps[0])
+    } else {
+        let mut iter = deps.iter();
+        let all_but_last = iter
+            .by_ref()
+            .take(deps.len() - 1)
+            .map(|s| format!("'{s}'",))
+            .collect::<Vec<_>>()
+            .join(", ");
+        let last = iter.next().unwrap();
+        format!("{all_but_last}, and '{last}'")
+    };
+
+    OxcDiagnostic::warn(if deps.len() == 1 {
+        format!("React Hook {hook_name} has a missing dependency: {deps_pretty}")
+    } else {
+        format!("React Hook {hook_name} has missing dependencies: {deps_pretty}")
+    })
+    .with_label(span)
+    .with_help("Either include it or remove the dependency array.")
+}
+
+fn unnecessary_dependency_diagnostic(hook_name: &str, dep_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("React Hook {hook_name} has unnecessary dependency: {dep_name}"))
+        .with_label(span)
+        .with_help("Either include it or remove the dependency array.")
+}
+
+fn dependency_array_not_array_literal_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "React Hook {hook_name} was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies."
+    ))
+    .with_label(span)
+    .with_help("Use an array literal as the second argument.")
+}
+
+fn literal_in_dependency_array_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("The literal is not a valid dependency because it never changes.")
+        .with_label(span)
```

Comment:
```suggestion
    OxcDiagnostic::warn("The literal is not a valid dependency because it never changes.")
        .with_help("Remove the literal from the array.")
        .with_label(span)
```